### DISCUSSION TITLE
Publish Livewire JS into it's own NPM package that can be included as a dependency for outside projects and frameworks

### DISCUSSION THREAD

**Discussion by grantcopley** [livewire/livewire#3568] [3. Feature ideas]
I am working on building a ColdBox module ( coldbox.org ) that uses the underlining Livewire JS functionality and libraries to bring Livewire's awesome functionality to CFML ( ColdFusion ). Because the Livewire JS is embedded within the repo as is, we would have to build and generate a JS artifact to include in the module, which would quickly become outdated.

Are there any plans or any chance in the world that the Livewire JS functionality could be its own NPM package that could be declared as a dependency for non-Laravel applications? This would ideally allow users in other frameworks and languages to stay updated with Livewire's JS functionality.

**Comment by lmajano** [comment]
💯

**Comment by joshhanley** [comment]
@grantcopley can you go into some detail about how you use the Livewire front end in your application?

Really curious to see how it's being used outside of Laravel 🙂

**Reply by grantcopley** [reply]
@joshhanley Sure thing. The Livewire JS assets from this repo are compiled down into a single file, and then referenced within the site's layout. Then it's simply a matter of matching the outgoing API request and responding with the correct responses that Livewire is expecting. The only difference is that the API responses are typically handled in Laravel and PHP, and instead, I'm returning the responses in CFML and ColdBox.

**Comment by calebporzio** [comment]
Very cool @grantcopley - No official plans - but I've started a rewrite of the JS for V3 of Livewire and making the JS decoupled from Laravel is a goal so far, so be on the lookout for the next release (won't be this year unfortunately)

**Comment by lmajano** [comment]
That's great news @calebporzio . PLease let us know how we can help in any way.

**Reply by calebporzio** [reply]
Will do, thanks!
